From ff53642c734d4b154bbd0ef2f7392a05ab89f3a4 Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Mon, 19 May 2014 16:56:01 +0800
Subject: [PATCH 4142/5965] Revert "Bug #92007. Fix vdec power on issue for TS
 when the vidoe format is not HEVC."

This reverts commit 4b53fb11cefcbfa506573be8c0513ecd82b6a281.
---
 drivers/amlogic/amports/amstream.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/amlogic/amports/amstream.c b/drivers/amlogic/amports/amstream.c
index e9b766ab32ba..6f97c89da08c 100755
--- a/drivers/amlogic/amports/amstream.c
+++ b/drivers/amlogic/amports/amstream.c
@@ -1088,8 +1088,7 @@ static int amstream_open(struct inode *inode, struct file *file)
         if (this->type & (PORT_TYPE_MPTS | PORT_TYPE_HEVC)) {
             vdec_poweron(VDEC_HEVC);
         }
-
-        if ((this->type & PORT_TYPE_HEVC) == 0) {
+        else if ((this->type & PORT_TYPE_HEVC) == 0) {
             vdec_poweron(VDEC_1);
         }
 #elif MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6TVD
-- 
2.19.0

